x-pack/plugins/apm/public/components/app/tutorials/instructions/api_key_callout.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiCallOut } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+
+export function ApiKeyCallout({
+  isSuccess,
+  isError,
+  errorMessage,
+}: {
+  isSuccess: boolean;
+  isError: boolean;
+  errorMessage?: string;
+}) {
+  if (isSuccess) {
+    return (
+      <>
+        <EuiCallOut
+          title={i18n.translate(
+            'xpack.apm.tutorial.apiKey.success.calloutTitle',
+            {
+              defaultMessage: 'API key created',
+            }
+          )}
+          color="success"
+          iconType="check"
+        >
+          {i18n.translate('xpack.apm.tutorial.apiKey.success.calloutMessage', {
+            defaultMessage: `Remember to store this information in a safe place. It won't be displayed anymore after you continue`,
+          })}
+        </EuiCallOut>
+      </>
+    );
+  }
+
+  const regex = /missing the following requested privilege\(s\)/;
+  const isWarning = isError && regex.test(errorMessage || '');
+
+  if (isWarning) {
+    return (
+      <EuiCallOut
+        title={i18n.translate(
+          'xpack.apm.tutorial.apiKey.warning.calloutTitle',
+          {
+            defaultMessage: 'User does not have permissions to create API Key',
+          }
+        )}
+        color="warning"
+        iconType="warning"
+      >
+        {i18n.translate('xpack.apm.tutorial.apiKey.warning.calloutMessage', {
+          defaultMessage:
+            'Please add the key generated by the admin in the dedicated space in the config below',
```

Comment:
This is worded in a way that makes it difficult to read. I think the actionable step here is to tell the user what permissions they need in order to produce an api key

```suggestion
            'You must have permission to <capability> to create api keys for APM agents.',
```

Reply:
The actual API response with the error, does mentions this, though not in a good format. But i found this message in the designs here - https://github.com/elastic/observability-dev/issues/2718#issuecomment-1557249182

Reply:
@sqren Since the API is not returning the capabilities, we can extract it out of the error message using a regex.

```
const errorMessage = 'editor is missing the following requested privilege(s): event:write.    You might try with the superuser, or add the missing APM application privileges to the role of the authenticated user, eg.:\n    PUT /_security/role/my_role\n    {\n      ...\n      \"applications\": [{\n        \"application\": \"apm\",\n        \"privileges\": [\"event:write\"],\n        \"resources\": [*]\n      }],\n      ...\n    }';

const privilegeRegex = /(?:missing the following requested privilege\(s\):\s)([\w:]+)/;
const match = errorMessage.match(privilegeRegex);
const missingPrivilege = match ? match[1] : null;
```

But this would make the implementation very fragile on the frontend.

Reply:
> But this would make the implementation very fragile on the frontend.

Yes, I'm not suggesting to extract the privileges out of the message. But we should know up front which privileges are required and list them.

> The actual API response with the error, does mentions this, though not in a good format. But i found this message in the designs here - https://github.com/elastic/observability-dev/issues/2718#issuecomment-1557249182

Okay, can you ask Isabella if this has been around a copy writer or if she is ok with us changing it? As mentioned, I think it could be worded better.  What does "in the dedicated space" refer to? Spaces are not available on serverless.

What does "Please add the key generated by the admin" mean? Has the admin already generated an api key? Or are we implying that the user should ask their admin to generate an admin on their behalf?

Reply:
HI @isaclfreire, i agree with Soren that may be we can list down the missing privilege as the only required privilege is `event:write` for API Key generation.

Reply:
Admin role will be required for initial serverless offering.  I'm working with the team to determine how we might modify that using available (Admin, Editor, Viewer) roles (if it is appropriate to do so).  In the future, custom roles should have an option for "Agent" Key generation.

Reply:
I don't think it's meaningful to list privs/permissions because they cannot be set in initial version of serverless.  It will be 100% role driven, so it will be most intuitive to direct to appropriate required role (Admin)

Reply:
@chrisdistasio thanks for confirming this.
In that case, i believe the current design which we have is fine to go with

Reply:
> In that case, i believe the current design which we have is fine to go with

Are you suggesting to keep the copy as-is? I don't think we should.

Reply:
> Admin role will be required for initial serverless offering. I'm working with the team to determine how we might modify that using available (Admin, Editor, Viewer) roles (if it is appropriate to do so). In the future, custom roles should have an option for "Agent" Key generation.

We should keep in mind that this flow will also be used for ESS and on-prem in the not too distant future where these limitations don't apply.

Reply:
@sqren ESS and onPrem will not use API Keys, they will be using secret token, which will be generated when the page will load and not with a button click

Reply:
> Admin role will be required for initial serverless offering. I'm working with the team to determine how we might modify that using available (Admin, Editor, Viewer) roles (if it is appropriate to do so). In the future, custom roles should have an option for "Agent" Key generation.

based on Chris's comment, i thought we can live with this existing message.

But since we already have the API change which could highlight the missing privileges, i am happy to change the text.

@sqren @isaclfreire  @chrisdistasio Any recommendations for the new copy ?

wdyt ? `User is missing following privileges - [LIST_OF_PRIVILEGES_COMMA SEPARATED]. Please check with your admin`

The second sentence can definitely be improved here.